src/java.base/share/classes/java/io/DataInputStream.java

```diff
@@ -579,7 +579,7 @@ public final String readUTF() throws IOException {
      *               valid modified UTF-8 encoding of a Unicode string.
      * @see        java.io.DataInputStream#readUnsignedShort()
      */
-    public static final String readUTF(DataInput in) throws IOException {
+    public static String readUTF(DataInput in) throws IOException {
```

Comment:
I remember a few years ago asked to create a CCC to remove the final keyword in the final class. This change seems broader, probably the rules are changed since then, but this one actually may affect the method signature. And subclasses will allow hiding this method.

Reply:
Now that you bring it up, a final modifier on static methods does show up in the javadoc, and is included in signature tests too.

@minborg I agree the final modifier doesn't make sense here but I think it will need a CSR to track the sig change.If you don't want to do that then just drop this part, and it can be done separately.